include/proxy-wasm/context.h

```diff
@@ -45,14 +45,16 @@ class WasmVm;
  * @param vm_id is a string used to differentiate VMs with the same code and VM configuration.
  * @param plugin_configuration is configuration for this plugin.
  * @param fail_open if true the plugin will pass traffic as opposed to close all streams.
+ * @param key is used for caching PluginHandleBase per thread and determining whether to create a
+ * new PluginHandleBase.
  */
 struct PluginBase {
   PluginBase(std::string_view name, std::string_view root_id, std::string_view vm_id,
-             std::string_view runtime, std::string_view plugin_configuration, bool fail_open)
+             std::string_view runtime, std::string_view plugin_configuration, bool fail_open,
+             std::string_view key)
       : name_(std::string(name)), root_id_(std::string(root_id)), vm_id_(std::string(vm_id)),
         runtime_(std::string(runtime)), plugin_configuration_(plugin_configuration),
-        fail_open_(fail_open), key_(root_id_ + "||" + plugin_configuration_),
-        log_prefix_(makeLogPrefix()) {}
+        fail_open_(fail_open), key_(key), log_prefix_(makeLogPrefix()) {}
```

Comment:
Should embedders control the complete plugin key or would it make sense to let them add an opaque value to the base plugin key instead, so that some uniquess that we know must be enforced would be done in the Proxy-Wasm C++ Host?

Reply:
the latter makes more sense to me, so I fixed accordingly. Thanks!